docs/specs/code/LOCK_WALLET_WITH_PIN.md

```diff
@@ -0,0 +1,41 @@
+# Abstract
+
+Allows user to lock their wallet with a pin like someone can do in iOS / Android.
+
+# Motivation
+
+Yoroi is completely secure (impossible to do tx without the password), but we want to add a lock screen similar to what phones have so we can protect the user privacy (balance and transactions) in case that someone else has access to the computer. This has a been a constant request from multiple people.
+
+# Proposal
+
+## How it should work
+
+The user should have like a possibility to click an icon in the topbar (a lock) and then the app will be locked; in the locked state, the user can’t see any information except for a dialog where a PIN is asked (we will name this pin/ pincode to avoid confusion with the spending password)
+
+## Requirements
+
+- add a dummy icon in the top right part of the topbar -> activate a locking screen
+- add a new item in the settings list with the option to enable / disable a locking screen (if it’s disable probably we should hide the icon on the top right bar) also besides a checkbox for this screen we should have: current password input / new password input / repeat new password input
+- when the user selects the checkbox to enable a password, we should show a dialog asking for a password and repeat password
+minimum 6 characters (max depending on the design). PIN will be stored as a hash + salt. Probably we should use SHA2, SHA3 or Blake2b.

```

Comment:
yeah, it’s wrong the PIN name. So probably it needs to be passcode or something like that. I think Shin recommended passcode.

Reply:
We could try popularise terms like: "Locking password" / "Spending password" 🤔

Reply:
@nicarq Apple's User Interface Guidelines are very specific about not using the term "passcode":
>Never use the term passcode. A passcode is used for unlocking the user's iOS device and authenticating with Apple Pay when biometric authentication is disabled.
(https://developer.apple.com/design/human-interface-guidelines/ios/user-interaction/authentication/)

Although we are currently discussing the feature for the extension, I think the terminology we use should be the same.

Reply:
@ebeloded yeah but as you said that’s for iOS. This is for web and I think this is even a reason to call it like that, because it already makes sense in the mobile ecosystem :)